src/test/golang/test_go_preempt.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"sync"
+	"time"
+)
+
+func busy(wg *sync.WaitGroup) {
+	t0 := time.Now()
+	t_finish := t0.Add(100 * time.Millisecond)
+	for time.Now().Before(t_finish) {
+		// Spin.
+
+		// For this test to pass, time must eventually move forward in such a
+		// spin loop.
+
+		// Really, though, this test is intended to test compatibility with
+		// golang's preemption mechanism, which will eventually interrupt with
+		// a signal to allow other goroutines to run.
```

Comment:
I don't know much about Go, but I don't see any other goroutines in this program. Does this test expect that the go runtime itself will be running its own goroutines, or should this test add a second goroutine to test this preemption?

Reply:
`main` starts 2 instances of this goroutine